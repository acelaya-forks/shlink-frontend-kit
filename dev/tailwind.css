@import 'tailwindcss' prefix(tw) important;
@import '../src/tailwind.preset.css';
