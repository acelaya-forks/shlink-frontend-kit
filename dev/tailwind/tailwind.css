@import 'tailwindcss' prefix(tw) important;
@import '../../src/tailwind/tailwind.preset.css';
