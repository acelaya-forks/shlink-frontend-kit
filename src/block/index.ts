export * from './Message';
export * from './Result';
export * from './SimpleCard';
