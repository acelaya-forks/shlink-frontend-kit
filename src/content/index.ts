export * from './Details';
export * from './Listbox';
export * from './Table';
