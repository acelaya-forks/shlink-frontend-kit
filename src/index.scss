@import './utils/ResponsiveTable';
@import './theme/theme';
