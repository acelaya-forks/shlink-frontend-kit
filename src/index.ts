export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
