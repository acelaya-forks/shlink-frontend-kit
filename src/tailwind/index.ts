export * from './content';
export * from './feedback';
export * from './form';
export * from './helpers';
export * from './navigation';
export * from './surfaces';
export * from './utils';
export * from './types';
