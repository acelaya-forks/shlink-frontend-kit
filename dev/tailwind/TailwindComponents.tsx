import type { FC } from 'react';
import { Route, Routes } from 'react-router';
import { TablePage } from './content/TablePage';
import { ModalDialogPage } from './feedback/ModalDialogPage';
import { ResultPage } from './feedback/ResultPage';
import { ButtonsPage } from './form/ButtonsPage';
import { InputsPage } from './form/InputsPage';
import { SearchInputPage } from './form/SearchInputPage';
import { NavPillsPage } from './navigation/NavPillsPage';
import { PaginatorPage } from './navigation/PaginatorPage';
import { CardsPage } from './surfaces/CardsPage';

export const TailwindComponents: FC = () => {
  return (
    <Routes>
      <Route path="/" element={<h2 className="text-center">Select component - Tailwind</h2>} />
      <Route path="/form/inputs" element={<InputsPage />} />
      <Route path="/form/search-input" element={<SearchInputPage />} />
      <Route path="/form/buttons" element={<ButtonsPage />} />
      <Route path="/surfaces/cards" element={<CardsPage />} />
      <Route path="/content/tables" element={<TablePage />} />
      <Route path="/navigation/paginator" element={<PaginatorPage />} />
      <Route path="/navigation/nav-pills">
        <Route path="" element={<NavPillsPage />} />
        <Route path="*" element={<NavPillsPage />} />
      </Route>
      <Route path="/feedback/dialogs" element={<ModalDialogPage />} />
      <Route path="/feedback/result" element={<ResultPage />} />
      <Route path="*" element={<h2 className="text-center">Not found - Tailwind</h2>} />
    </Routes>
  );
};
