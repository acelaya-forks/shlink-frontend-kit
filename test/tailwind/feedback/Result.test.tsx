import { render } from '@testing-library/react';
import type { ResultProps } from '../../../src/tailwind';
import { Result } from '../../../src/tailwind';
import { checkAccessibility } from '../../__helpers__/accessibility';

describe('<Result />', () => {
  const setUp = (props: Partial<ResultProps> = {}) => render(<Result variant="success" {...props} />);

  it('passes a11y checks', () => checkAccessibility(setUp({ children: 'Message content' })));

  it.each([
    {},
    { size: 'sm' as const },
    { size: 'md' as const },
    { size: 'lg' as const },
    { variant: 'success' as const },
    { variant: 'error' as const },
    { variant: 'warning' as const },
  ])('renders as expected based on provided props', (props) => {
    const { container } = setUp(props);
    expect(container).toMatchSnapshot();
  });
});
