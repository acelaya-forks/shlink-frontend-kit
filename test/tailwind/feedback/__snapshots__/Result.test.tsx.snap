// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Result /> > renders as expected based on provided props 1`] = `
<div>
  <div
    class="tw:rounded-md tw:text-center tw:p-4 tw:[&]:text-white tw:bg-brand"
  />
</div>
`;

exports[`<Result /> > renders as expected based on provided props 2`] = `
<div>
  <div
    class="tw:rounded-md tw:text-center tw:p-2 tw:[&]:text-white tw:bg-brand"
  />
</div>
`;

exports[`<Result /> > renders as expected based on provided props 3`] = `
<div>
  <div
    class="tw:rounded-md tw:text-center tw:p-4 tw:[&]:text-white tw:bg-brand"
  />
</div>
`;

exports[`<Result /> > renders as expected based on provided props 4`] = `
<div>
  <div
    class="tw:rounded-md tw:text-center tw:p-6 tw:[&]:text-white tw:bg-brand"
  />
</div>
`;

exports[`<Result /> > renders as expected based on provided props 5`] = `
<div>
  <div
    class="tw:rounded-md tw:text-center tw:p-4 tw:[&]:text-white tw:bg-brand"
  />
</div>
`;

exports[`<Result /> > renders as expected based on provided props 6`] = `
<div>
  <div
    class="tw:rounded-md tw:text-center tw:p-4 tw:[&]:text-white tw:bg-danger"
  />
</div>
`;

exports[`<Result /> > renders as expected based on provided props 7`] = `
<div>
  <div
    class="tw:rounded-md tw:text-center tw:p-4 tw:bg-warning tw:text-black"
  />
</div>
`;
